bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,295 @@
+        }
+        updateStatus();
+    }
+
+    private void updateStatus() {
+        CBusNetworkConfiguration configuration = this.configuration;
+        if (configuration == null) {
+            logger.debug("updateStatus - NetworkHandler not initialised");
+            return;
+        }
+        ThingStatus lastStatus = getThing().getStatus();
+        Network network = getNetwork();
+        CBusCGateHandler cbusCGateHandler = getCBusCGateHandler();
+        try {
+            if (cbusCGateHandler == null || !cbusCGateHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "CGate connection offline");
+            } else if (network == null) {
+                logger.debug("No network - set configuration error");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No Network object available");
+            } else if (network.isOnline()) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Network is not reporting online");
+            }
+        } catch (CGateException e) {
+            logger.warn("Problem checking network state for network {}", network.getNetworkID(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        if (!getThing().getStatus().equals(lastStatus)) {
+            ScheduledFuture<?> networkSync = this.networkSync;
+            if (lastStatus == ThingStatus.OFFLINE) {
+                if (networkSync == null || networkSync.isCancelled()) {
+                    this.networkSync = scheduler.scheduleWithFixedDelay(this::doNetworkSync, 10,
+                            configuration.syncInterval, TimeUnit.SECONDS);
+                }
+            } else {
+                if (networkSync != null) {
+                    networkSync.cancel(false);
+                }
+            }
+            for (Thing thing : getThing().getThings()) {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof CBusGroupHandler) {
+                    ((CBusGroupHandler) handler).updateStatus();
+                }
+            }
+        }
+    }
+
+    private void doNetworkSync() {
+        Network network = getNetwork();
+        try {
+            if (getThing().getStatus().equals(ThingStatus.ONLINE) && network != null) {
+                logger.info("Starting network sync on network {}", network.getNetworkID());
+                network.startSync();
+            }
+        } catch (CGateException e) {
+            logger.warn("Cannot start network sync on network {} - {}", network.getNetworkID(), e.getMessage());
+        }
+    }
+
+    public synchronized @Nullable CBusCGateHandler getCBusCGateHandler() {
```

Comment:
You don't need to cache the CBusCGateHandler returned from this method, you can just call this method everytime you need the handler. Please remove synchronized from this method and get rid of the `bridgeHandler` field.

Reply:
Updated and made it private as well.